packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/types/AnyType.kt

```diff
@@ -44,6 +65,86 @@ class LazyKType(
   }
 }
 
+class EmptyKType(
+  override val classifier: KClass<*>,
+  override val isMarkedNullable: Boolean = false
+) : KType {
+  override val annotations: List<Annotation>
+    get() = emptyList()
+  override val arguments: List<KTypeProjection>
+    get() = emptyList()
+
+  override fun equals(other: Any?): Boolean {
+    if (this === other) return true
+    if (other !is EmptyKType) return false
+
+    return classifier == other.classifier && isMarkedNullable == other.isMarkedNullable
+  }
+
+  override fun hashCode(): Int {
+    return 31 * classifier.hashCode() + isMarkedNullable.hashCode()
```

Comment:
nit: maybe comment why it's 31 here

Reply:
That's a good question. To be honest, I have never thought about that. The number `31` is generally used by Java programmers. Also, it's in the generated version of the `hashCode` function by Android Studio.

Reply:
Here's an explanation: 
```
The value 31 was chosen because it is an odd prime. If it were even and the multiplication overflowed, information would be lost, as multiplication by 2 is equivalent to shifting. The advantage of using a prime is less clear, but it is traditional. A nice property of 31 is that the multiplication can be replaced by a shift and a subtraction for better performance: 31 * i == (i << 5) - i. Modern VMs do this sort of optimization automatically.
``

Reply:
> A nice property of 31 is that the multiplication can be replaced by a shift and a subtraction for better performance: 31 * i == (i << 5) - i. Modern VMs do this sort of optimization automatically.

that is a great explanation. thanks for sharing that.